free-1.md

```diff
@@ -57,12 +98,30 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+I would guess that quite a bit of time would go into figuring out exactly how
+the user should interact with the language. This language would be some
+combination of formal logic, general-purpose programming language, and English.
+It needs to be easy to use but also extremely general --- it would be bad if a
+game _almost_ fits in the language because then the language becomes useless.
+Programming the interpreter for the language would also be difficult but
+probably not as bad. If the language is closer to formal logic, some basic
+translation could happen and then be fed into a logic prover. Outputting game
+rules would be complex if one wanted them to look good but simple to get
+something basic done.
 
 ### Scope
 _How big an idea is this? How ambitious is this project?_
 
+This seems like a fairy ambitious project. Both the design and implementation
+sound pretty difficult and I'm still not sure if things would even work.
 
 ### Benefits and drawbacks
 _Why might this be a good idea for a project? Why might this not be a good idea 
 project?_
 
+This is a good idea for a project because it's clearly a place where a DSL is
+good and could be used for a wide variety of games. Its disadvantages are that
+it's probably not the most useful --- any text that it outputs would probably
+be awkward enough that a game designer would reword it, and that could easily
+reintroduce the very "bugs" that the system is designed to remove.
+
```

Comment:
"Games" is a good domain, but a large one. It wasn't clear at first, but I think you're proposing to make it possible to specify the rules of games and to check those rules for...consistency? Loopholes? Ambiguity? It's not quite clear what would happen when a program runs. Specific examples would help make things clearer, going forward.

Still, I think this is an interesting project idea.  If you go with it, I recommend a bit of research to tie particular kinds of games to, e.g., particular kinds of computable logic. For example, do partial-information games (such as poker) correspond to a particular kind of logic of notion of computability? Also, it might be good at first to limit the domain to a particular style of game, e.g., cards, dice, or boards, then extend the language to handle other styles of games.

You might want to check out [game semantics](https://www.dpmms.cam.ac.uk/~martin/Research/Oldpapers/gamesemantics97scan.pdf). It's more about game theory than games _per se_, but you might find it interesting. There's also a book called [Games, Puzzles, and Computation](http://erikdemaine.org/papers/GPC/).

As a particularly specific suggestion: Could you encode the rules of the logic-based game [Hanabi](https://en.wikipedia.org/wiki/Hanabi_(card_game)), such that a program would suggest the best course of action for each turn (including which hint provides the most information)?